pkg/common/cns-lib/volume/util.go

```diff
@@ -74,3 +77,21 @@ func IsDiskAttachedToVMs(ctx context.Context, volumeID string, vms []*cnsvsphere
 	}
 	return "", nil
 }
+
+// updateQueryResult helps update CnsQueryResult to populate volume.Metadata.EntityMetadata.ClusterID
+// with value from volume.Metadata.ContainerCluster.ClusterId
+// This is required to make driver code compatible to vSphere 67 release
+func updateQueryResult(m *defaultManager, res *cnstypes.CnsQueryResult) *cnstypes.CnsQueryResult {
+	if m.virtualCenter.Client.Version == cns.ReleaseVSAN67u3 {
+		for volumeIndex, volume := range res.Volumes {
+			for metadataIndex, metadata := range volume.Metadata.EntityMetadata {
+				if metadata != nil {
+					metadata.(*cnstypes.CnsKubernetesEntityMetadata).ClusterID = volume.Metadata.ContainerCluster.ClusterId
```

Comment:
Can you check the type of metadata is cnstypes.CnsKubernetesEntityMetadata before assigning the ClusterId?

Reply:
done.

```
if cnsK8sMetaEntityMetadata, ok := metadata.(*cnstypes.CnsKubernetesEntityMetadata); ok {
	cnsK8sMetaEntityMetadata.ClusterID = volume.Metadata.ContainerCluster.ClusterId
	volume.Metadata.EntityMetadata[metadataIndex] = cnsK8sMetaEntityMetadata
} else {
	log.Debugf("metadata: %v is not of type CnsKubernetesEntityMetadata", metadata)
}
```